File: coj.uci/m2946.cs
class m2946
{
public static void Main(string[] args)
{
int n = int.Parse(System.Console.ReadLine());
int i = 0;
while(n-->0)
{
	i = int.Parse(System.Console.ReadLine());
	if ((i&1)==1) System.Console.WriteLine(i<<1);
	else
	{
	while(i>0 && (i%2==0)) i/=2;
	System.Console.WriteLine(i);
	}
}
}
}

File: coj uci/src/m2946.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

class m2946 {
    static int[] b = new int[50000];
    static String newline = System.lineSeparator();

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = br.readLine();
        int n = Integer.parseInt(line);
        int i;
        setVals();
        StringBuilder sb = new StringBuilder();

        while (n-- > 0) {
            line = br.readLine();
            i = Integer.parseInt(line);
            if ((i & 1) == 1) sb.append((i << 1)).append(newline);
            else sb.append(b[i]).append(newline);
        }
        System.out.println(sb);
    }

    public static void setVals() {
        for (int i = 1; i < 50000; i += 2) {
            b[i] = i;
        }
        for (int i = 2; i < 50000; i += 2) {
            b[i] = b[i >> 1];
        }
    }

}